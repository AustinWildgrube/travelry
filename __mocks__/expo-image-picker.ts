module.exports = {
  __esModule: true,
  launchImageLibraryAsync: jest.fn().mockReturnValue({
    canceled: false,
    assets: [
      {
        uri: 'file://test-uri',
      },
    ],
  }),
  MediaTypeOptions: {
    All: 'all',
    Videos: 'videos',
    Images: 'images',
  },
};
