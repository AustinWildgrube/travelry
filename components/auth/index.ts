export * from './RegistrationForm';
export * from './SetupConfetti';
export * from './SetupProfileDetails';
export * from './SetupProfileImage';