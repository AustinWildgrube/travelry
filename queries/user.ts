import { Alert } from 'react-native';

import { decode } from 'base64-arraybuffer';
import { CryptoDigestAlgorithm, digestStringAsync } from 'expo-crypto';
import { type ImageResult } from 'expo-image-manipulator';

import { createAvatarsSignedUploadUrl } from '&/queries/signature';
import { supabase } from '&/services/supabase';
import { type User, type UserFollows, type UserSummary } from '&/types/types';

export const getUserAccount = async (id: string, currentUserId: string): Promise<User> => {
  // TODO: change cover photo to be anything not just a post media
  // TODO: allow for post_count to increment
  const { data, error } = await supabase
    .from('account')
    .select(
      `
      id,
      username, 
      first_name,
      last_name,
      avatar_url,
      avatar_placeholder,
      bio,
      account_stat (
        following_count, 
        followers_count, 
        trip_count
      ),
      album (
        id,
        name,
        cover,
        cover_placeholder,
        post_count
      ),
      is_following:follow!target_account_id(count)
    `,
    )
    .eq('id', id)
    .eq('follow.account_id', currentUserId)
    .eq('follow.target_account_id', id)
    .order('name', { foreignTable: 'album', ascending: true })
    .single();

  if (error) throw new Error(`getUser(${error.code}): ${error.message}`);

  // transform is_following from [{ count: 0 }] to boolean
  const user = data as User;
  const userCount = user.is_following as { count: number }[];
  user.is_following = userCount[0].count > 0;

  console.log(user);

  return user;
};

export const getUserAccountSearch = async (searchText: string): Promise<UserSummary[]> => {
  const { data: conversationRecipients, error } = await supabase
    .from('account')
    .select(
      `
      id,
      username,
      first_name,
      last_name,
      avatar_url,
      avatar_placeholder
    `,
    )
    .or(`username.ilike.%${searchText}%,first_name.ilike.%${searchText}%,last_name.ilike.%${searchText}%`);

  if (error) throw new Error(`getUserAccountSearch(${error.code}): ${error.message}`);

  return conversationRecipients as UserSummary[];
};

export const updateUserAccount = async (id: string, updates: Partial<User>): Promise<void> => {
  const { error } = await supabase.from('account').update(updates).eq('id', id);
  if (error) throw new Error(`updateUserAccount(${error.code}): ${error.message}`);
};

export const uploadUserAvatar = async (userId: string, avatarImage: ImageResult): Promise<void> => {
  if (!avatarImage.base64) return;

  const hash = await digestStringAsync(CryptoDigestAlgorithm.SHA256, avatarImage.uri);
  const filename = `${hash}.jpg`;

  const signedUrl = await createAvatarsSignedUploadUrl(filename);
  const { error } = await supabase.storage.from('avatars').uploadToSignedUrl(filename, signedUrl, decode(avatarImage.base64), {
    upsert: true,
    contentType: 'image/jpeg',
  });

  if (error) {
    if (error.statusCode === 413) Alert.alert('Image is too large', 'Please select an image under 2MB.');
    throw new Error(`uploadUserAvatarOne(${error.statusCode}): ${error.message}`);
  }

  const { error: errorTwo } = await supabase.from('account').update({ avatar_url: filename }).eq('id', userId);
  if (errorTwo) throw new Error(`uploadUserAvatarTwo(${errorTwo.code}): ${errorTwo.message}`);
};

export const followUser = async (id: string, targetId: string): Promise<void> => {
  let { error } = await supabase.from('follow').insert({ account_id: id, target_account_id: targetId });
  if (error) throw new Error(`followUser(${error.code}): ${error.message}`);
};

export const unfollowUser = async (id: string, targetId: string): Promise<void> => {
  let { error } = await supabase.from('follow').delete().eq('account_id', id).eq('target_account_id', targetId);
  if (error) throw new Error(`unfollowUser(${error.code}): ${error.message}`);
};

export const getUserFollowers = async (id: string): Promise<UserFollows[]> => {
  const { data, error } = await supabase
    .from('follow')
    .select(
      `
        id,
        account!account_id (
          id,
          username,
          first_name,
          last_name,
          avatar_url,
          avatar_placeholder
        ),
        is_following:account!target_account_id(count)
      `,
    )
    .eq('target_account_id', id);

  if (error) throw new Error(`getUserFollowers(${error.code}): ${error.message}`);

  // transform is_following from [{ count: 0 }] to boolean
  const followers = data as UserFollows[];
  for (let i = 0; i < followers.length; i++) {
    if (!followers[i].is_following) {
      followers[i].is_following = false;
    } else {
      // @ts-ignore
      followers[i].is_following = followers[i].is_following.count > 0;
    }
  }

  return followers as UserFollows[];
};

export const getUserFollowing = async (id: string): Promise<UserFollows[]> => {
  const { data, error } = await supabase
    .from('follow')
    .select(
      `
        id,
        account!target_account_id (
          id,
          username,
          first_name,
          last_name,
          avatar_url,
          avatar_placeholder
        ),
        is_following:account!target_account_id(count)
      `,
    )
    .eq('account_id', id);

  if (error) throw new Error(`getUserFollowing(${error.code}): ${error.message}`);

  // transform is_following from [{ count: 0 }] to boolean
  const followers = data as UserFollows[];
  for (let i = 0; i < followers.length; i++) {
    if (!followers[i].is_following) {
      followers[i].is_following = false;
    } else {
      // @ts-ignore
      followers[i].is_following = followers[i].is_following.count > 0;
    }
  }

  return data as UserFollows[];
};
