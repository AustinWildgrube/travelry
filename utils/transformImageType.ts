import { manipulateAsync, type ImageResult, type SaveFormat } from 'expo-image-manipulator';

export const transformImageType = async (imageUri: string, type: SaveFormat): Promise<ImageResult> => {
  return await manipulateAsync(imageUri, [], { compress: 0.5, format: type, base64: true });
};
