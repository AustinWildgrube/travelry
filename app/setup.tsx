import { Fragment, useState } from 'react';

import { Link } from 'expo-router';
import { SafeAreaView } from 'react-native-safe-area-context';

import { Confetti, SetupProfileDetails, SetupProfileImage } from '&/components/auth';
import { Button, Heading, Text } from '&/components/core';
import { CONFETTI_FALL_SPEED, CONFETTI_ITEM_HEIGHT, CONFETTI_ITEM_WIDTH } from '&/utils/constants';
import { makeStyles } from '&/utils/makeStyles';

export default function Setup(): JSX.Element {
  const styles = useStyles();
  const [step, setStep] = useState(0);

  return (
    <SafeAreaView style={styles.setup}>
      <Heading>Setup your profile</Heading>

      {step === 0 ? (
        <Fragment>
          <Text style={styles.setupSlogan}>Upload an image - it's worth a thousand words</Text>
          <SetupProfileImage setStep={setStep} />

          <Button onPress={() => setStep(1)} variant="link" style={styles.skipLink}>
            <Button.Text>Skip</Button.Text>
          </Button>

          <Confetti
            fallSpeed={CONFETTI_FALL_SPEED}
            flipSpeed={3}
            horizSpeed={50}
            itemColors={['#84a98c', '#ffb703', '#023047']}
            itemDimensions={{ height: CONFETTI_ITEM_HEIGHT, width: CONFETTI_ITEM_WIDTH }}
            itemTintStrength={0.8}
            numItems={100}
          />
        </Fragment>
      ) : (
        <Fragment>
          <Text style={styles.setupSlogan}>Tell the world a little about yourself</Text>
          <SetupProfileDetails />

          <Link href="/feed" asChild>
            <Button variant="link" style={styles.skipLink}>
              <Button.Text>Skip</Button.Text>
            </Button>
          </Link>
        </Fragment>
      )}
    </SafeAreaView>
  );
}

const useStyles = makeStyles(theme => ({
  setup: {
    flex: 1,
    padding: theme.space['4'],
  },
  setupSlogan: {
    marginTop: theme.space['1'],
    marginBottom: theme.space['6'],
  },
  skipLink: {
    marginTop: theme.space['3'],
    zIndex: -1,
  },
}));
