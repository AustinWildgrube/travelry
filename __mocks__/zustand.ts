import { act } from '@testing-library/react-native';
import { StateCreator } from 'zustand';

import { useUserStore } from '&/stores/user';
import { type User } from '&/types/types';

const { create: actualCreate } = jest.requireActual('zustand');
const storeResetFns = new Set<() => void>();

export const mockUser: User = {
  id: '1',
  username: '',
  first_name: '',
  last_name: '',
  avatar_url: '',
  avatar_placeholder: '',
  is_following: false,
  bio: '',
  account_stat: {
    followers_count: 0,
    following_count: 0,
    trip_count: 0,
  },
  album: [
    {
      id: '',
      name: '',
      cover: '',
      cover_placeholder: '',
      post_count: 0,
    },
  ],
};

export const create = <S>(createState: StateCreator<S>) => {
  const store = actualCreate(createState);
  const initialState = store.getState();

  storeResetFns.add(() => {
    store.setState(initialState, true);
  });

  return store;
};

beforeEach(async () => {
  await act(() => {
    storeResetFns.forEach(resetFn => {
      resetFn();
    });

    useUserStore.setState({ setUser: undefined, updateUserStats: undefined, user: mockUser });
  });
});
