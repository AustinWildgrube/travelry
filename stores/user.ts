import { create } from 'zustand';

import { type User, type UserStat } from '&/types/types';

type UserState = {
  user: User | null;
  setUser: (user: User) => void;
  updateUserStats: (stats: UserStat) => void;
};

export const useUserStore = create<UserState>(set => ({
  user: null,
  setUser: (user: User) => {
    set({ user });
  },
  updateUserStats: (stats: UserStat) => {
    set(state => ({
      user: state.user
        ? {
            ...state.user,
            account_stat: {
              ...state.user.account_stat,
              ...stats,
            },
          }
        : null,
    }));
  },
}));
