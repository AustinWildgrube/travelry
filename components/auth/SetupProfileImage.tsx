import { useState, type Dispatch, type SetStateAction } from 'react';

import { type CameraCapturedPicture } from 'expo-camera';
import { SaveFormat } from 'expo-image-manipulator';
import { launchImageLibraryAsync, MediaTypeOptions, type ImagePickerAsset } from 'expo-image-picker';
import { User } from 'lucide-react-native';

import { Camera } from '&/components/camera/Camera';
import { Avatar, Button, Icon, Spinner, VStack } from '&/components/core';
import { LoadingIndicator } from '&/components/shared/LoadingIndicator';
import { uploadUserAvatar } from '&/queries/user';
import { useUserStore } from '&/stores/user';
import { makeStyles } from '&/utils/makeStyles';
import { transformImageType } from '&/utils/transformImageType';

type SetupProfileProps = {
  setStep: Dispatch<SetStateAction<number>>;
};

export function SetupProfileImage({ setStep }: SetupProfileProps): JSX.Element {
  const styles = useStyles();

  const [showCamera, setShowCamera] = useState(false);
  const [imageUploading, setImageUploading] = useState(false);
  const [imageUri, setImageUri] = useState<string | null>(null);

  const user = useUserStore(state => state.user);
  if (!user) return <LoadingIndicator />;

  const pickAvatarImage = async (): Promise<void> => {
    let result = await launchImageLibraryAsync({
      mediaTypes: MediaTypeOptions.Images,
      allowsEditing: true,
      aspect: [1, 1],
      quality: 1,
      exif: true,
    });

    if (!result.canceled) {
      onSuccess(result.assets[0]);
    }
  };

  const onSuccess = (image: CameraCapturedPicture | ImagePickerAsset): void => {
    setImageUploading(true);

    transformImageType(image.uri, SaveFormat.JPEG).then(async transformedImage => {
      setImageUri(transformedImage.uri);
      await uploadUserAvatar(user.id, transformedImage);
      setStep(1);
    });
  };

  return (
    <VStack space="lg" style={styles.setupProfile}>
      {imageUri ? (
        <Avatar style={styles.avatar}>
          <Avatar.Image source={{ uri: imageUri }} />
        </Avatar>
      ) : (
        <Avatar style={styles.avatar}>
          <Icon as={User} style={styles.avatarIcon} />
        </Avatar>
      )}

      <Button onPress={pickAvatarImage} isDisabled={imageUploading} size="xl">
        {imageUploading && <Spinner style={styles.buttonSpinner} />}
        <Button.Text>Upload an Image</Button.Text>
      </Button>

      <Button onPress={() => setShowCamera(true)} isDisabled={imageUploading} size="xl" variant="outline">
        {imageUploading && <Spinner style={styles.buttonSpinner} />}
        <Button.Text>Take an Image</Button.Text>
      </Button>

      {showCamera && <Camera onSuccess={onSuccess} setIsShown={setShowCamera} />}
    </VStack>
  );
}

const useStyles = makeStyles(theme => ({
  setupProfile: {
    flex: 1,
    justifyContent: 'center',
  },
  avatar: {
    borderRadius: theme.radii['full'],
    height: theme.space['56'],
    width: theme.space['56'],
  },
  avatarIcon: {
    color: theme.colors['white'],
    height: theme.space['56'],
    width: theme.space['56'],
  },
  buttonSpinner: {
    color: theme.colors['white'],
    marginRight: theme.space['1'],
  },
}));
