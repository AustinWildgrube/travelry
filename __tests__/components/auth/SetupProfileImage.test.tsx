import { fireEvent, screen } from '@testing-library/react-native';
import { MediaTypeOptions, launchImageLibraryAsync } from 'expo-image-picker';

import { SetupProfileImage } from '&/components/auth/SetupProfileImage';
import { uploadUserAvatar } from '&/queries/user';
import { transformImageType } from '&/utils/transformImageType';

import { mockUser } from '&/__mocks__/zustand';
import { render } from '&/jest.setup';

jest.mock('&/utils/transformImageType', () => ({
  transformImageType: jest.fn().mockResolvedValue({ uri: 'transformedImageUri' }),
}));

jest.mock('&/queries/user', () => ({
  uploadUserAvatar: jest.fn().mockResolvedValue({}),
}));

const setStepMock = jest.fn();

describe('SetupProfileImage', () => {
  beforeEach(() => {
    render(<SetupProfileImage setStep={setStepMock} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    screen.toJSON();
    expect(screen).toMatchSnapshot();
  });

  it('should render both buttons', () => {
    expect(screen.getByRole('button', { name: /Upload an Image/i })).toBeDefined();
    expect(screen.getByRole('button', { name: /Take an Image/i })).toBeDefined();
  });

  it('should allow an image to be uploaded from the users library', async () => {
    fireEvent.press(screen.getByRole('button', { name: /Upload an Image/i }));

    expect(launchImageLibraryAsync).toHaveBeenCalledWith({
      mediaTypes: MediaTypeOptions.Images,
      allowsEditing: true,
      aspect: [1, 1],
      quality: 1,
      exif: true,
    });

    expect(await screen.findByRole('button', { name: /Upload an Image/i })).toBeDisabled();
    expect(transformImageType).toHaveBeenCalledWith('file://test-uri', 'jpeg');
    expect(uploadUserAvatar).toHaveBeenCalledWith(mockUser.id, { uri: 'transformedImageUri' });
    expect(setStepMock).toHaveBeenCalledWith(1);
  });

  it('should allow an image to be taken with the phone camera', async () => {
    fireEvent.press(screen.getByRole('button', { name: /Take an Image/i }));
    expect(await screen.findByTestId('camera')).toBeDefined();

    // TODO: figure out why onPictureSaved is not being called
  });
});
