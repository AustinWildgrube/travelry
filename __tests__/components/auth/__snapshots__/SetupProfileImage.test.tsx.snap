// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SetupProfileImage should match the snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#ffffff",
        "flex": 1,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        replaceTheme={[Function]}
        style={
          {
            "backgroundColor": "#bdc6cf",
            "borderRadius": 110,
            "height": 220,
            "marginBottom": 24,
            "width": 220,
          }
        }
        theme={
          {
            "colors": {
              "background": "#ffffff",
              "black": "#242424",
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "inputBackground": "#fff",
              "inputBorder": "#d0d5dd",
              "inputLabel": "#242424",
              "platform": {
                "android": {
                  "error": "#f44336",
                  "grey": "rgba(0, 0, 0, 0.54)",
                  "primary": "#2196f3",
                  "searchBg": "#dcdce1",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "default": {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "ios": {
                  "error": "#ff3b30",
                  "grey": "#7d7d7d",
                  "primary": "#007aff",
                  "searchBg": "#dcdce1",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
                "web": {
                  "error": "#ff190c",
                  "grey": "#393e42",
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#ca71eb",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              },
              "primary": "#84a98c",
              "primary2": "#6A8770",
              "searchBg": "#303337",
              "secondary": "#ad1457",
              "success": "#52c41a",
              "warning": "#faad14",
              "white": "#ffffff",
            },
            "mode": "light",
            "spacing": {
              "lg": 12,
              "md": 8,
              "sm": 4,
              "xl": 24,
              "xs": 2,
            },
          }
        }
        updateTheme={[Function]}
      >
        <View
          accessibilityIgnoresInvertColors={true}
          style={
            {
              "backgroundColor": "transparent",
              "borderRadius": 110,
              "flex": 1,
              "overflow": "hidden",
              "position": "relative",
            }
          }
        >
          <Image
            borderRadius={110}
            onLoad={[Function]}
            style={
              {
                "bottom": 0,
                "flex": 1,
                "height": undefined,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": undefined,
              }
            }
            testID="RNE__Avatar__Image"
            transitionDuration={360}
          />
          <View
            accessibilityElementsHidden={false}
            collapsable={false}
            importantForAccessibility="yes"
            pointerEvents="auto"
            style={
              {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "height": undefined,
                  "justifyContent": "center",
                  "width": undefined,
                }
              }
              testID="RNE__Image__placeholder"
            >
              <View
                style={
                  {
                    "overflow": "hidden",
                  }
                }
                testID="RNE__ICON__CONTAINER"
              >
                <View
                  accessibilityRole="button"
                  android_ripple={
                    {
                      "borderless": false,
                      "color": "rgba(0, 0, 0, 0.3)",
                      "radius": -5,
                    }
                  }
                  disabled={false}
                  style={{}}
                  testID="RNE__ICON__CONTAINER_ACTION"
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                    testID="RNE__ICON"
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        [
                          {
                            "color": "white",
                            "fontSize": 110,
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                          {
                            "fontFamily": "feather",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                      testID="RNE__ICON__Component"
                    >
                      ïˆ„
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              {
                "flex": 1,
                "height": undefined,
                "width": undefined,
              }
            }
            testID="RNE__Image__children__container"
          />
        </View>
      </View>
      <View
        style={
          {
            "gap": 12,
            "width": "100%",
          }
        }
      >
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
              undefined,
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "#84a98c",
                  "borderBottomWidth": 4,
                  "borderColor": "#6A8770",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 12,
                  "paddingHorizontal": 14,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#fff",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Upload an Image
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            [
              {
                "overflow": "hidden",
              },
              {
                "borderRadius": 8,
              },
              undefined,
              false,
            ]
          }
          testID="RNE_BUTTON_WRAPPER"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": false,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 4,
                  "borderColor": "#6A8770",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "padding": 12,
                  "paddingHorizontal": 14,
                }
              }
            >
              <Text
                style={
                  {
                    "color": "#84a98c",
                    "fontSize": 18,
                    "fontWeight": "600",
                    "paddingVertical": 1,
                    "textAlign": "center",
                  }
                }
              >
                Take an Image
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
